### DISCUSSION TITLE
如何使用get_bot？

### DISCUSSION THREAD

**Discussion by Cranyozen** [nonebot/discussions#213] [问答] [answered]
我想实现主动发消息，我新建了一个定时任务。
但执行get_bot()报错`ValueError: There are no bots to get.`
执行get_bots()返回{}
这。。我很疑惑。。。

**Comment by yanyongyu** [comment, marked as answer]
你的bot没有连接上自然就没有bot，get不到

**Reply by Cranyozen** [reply]
Soga

**Reply by zhtoatao** [reply]
我想问一下怎么样才算连接上了......我后端用的lagrange，写其他的响应没问题，就是get_bot()不行，没办法主动发信息

**Reply by yanyongyu** [reply]
连接上了会有日志显示，通常是在ws accept的时候

**Reply by zhtoatao** [reply]
11-27 18:59:28 [INFO] uvicorn | ('127.0.0.1', 51154) - "WebSocket /onebot/v11/ws" [accepted]
是这个吗，看样子是收到了，但是我在plugin里写get_bot()的时候报错ValueError: There are no bots to get，吧self_id写进去也是同样的报错

**Reply by yanyongyu** [reply]
应该是bot xxx connected的日志
